client/components/table/index.js

```diff
@@ -67,33 +93,111 @@ class TableCard extends Component {
 		return rows.map( row => row.filter( ( col, i ) => showCols[ i ] ) );
 	}
 
+	selectAllRows( event ) {
+		const { ids } = this.props;
+		if ( event.target.checked ) {
+			this.setState( {
+				selectedRows: ids,
+			} );
+		} else {
+			this.setState( {
+				selectedRows: [],
+			} );
+		}
+	}
+
+	selectRow( i, event ) {
+		const { ids } = this.props;
+		if ( event.target.checked ) {
+			this.setState( ( { selectedRows } ) => ( {
+				selectedRows: uniq( [ ids[ i ], ...selectedRows ] ),
+			} ) );
+		} else {
+			this.setState( ( { selectedRows } ) => {
+				const index = selectedRows.indexOf( ids[ i ] );
+				return {
+					selectedRows: [ ...selectedRows.slice( 0, index ), ...selectedRows.slice( index + 1 ) ],
+				};
+			} );
+		}
+	}
+
+	getCheckbox( i ) {
+		const { ids = [] } = this.props;
+		const { selectedRows } = this.state;
+		const isChecked = -1 !== selectedRows.indexOf( ids[ i ] );
+		return {
+			display: (
+				<input type="checkbox" onChange={ partial( this.selectRow, i ) } checked={ isChecked } />
+			),
+			value: false,
+		};
+	}
+
+	getAllCheckbox() {
+		const { ids = [] } = this.props;
+		const { selectedRows } = this.state;
+		const isAllChecked = ids.length === selectedRows.length;
+		return {
+			label: (
+				<input
+					type="checkbox"
+					onChange={ this.selectAllRows }
+					aria-label={ __( 'Select All' ) }
+					checked={ isAllChecked }
+				/>
+			),
+			required: true,
+		};
+	}
+
 	render() {
 		const {
+			compareBy,
 			onClickDownload,
 			onQueryChange,
 			query,
 			rowHeader,
+			rowsPerPage,
 			summary,
 			title,
 			totalRows,
-			rowsPerPage,
 		} = this.props;
 		const { showCols } = this.state;
 		const allHeaders = this.props.headers;
-		const headers = this.filterCols( this.props.headers );
-		const rows = this.filterCols( this.props.rows );
+		let headers = this.filterCols( this.props.headers );
+		let rows = this.filterCols( this.props.rows );
+		if ( compareBy ) {
+			rows = rows.map( ( row, i ) => {
+				return [ this.getCheckbox( i ), ...row ];
+			} );
+			headers = [ this.getAllCheckbox(), ...headers ];
+		}
+
+		const className = classnames( {
+			'woocommerce-table': true,
+			'has-compare': !! compareBy,
+		} );
 
 		return (
 			<Card
-				className="woocommerce-table"
+				className={ className }
 				title={ title }
-				action={
+				action={ [
+					compareBy && (
+						<Button onClick={ this.onCompare } isDefault>
+							{ __( 'Compare', 'wc-admin' ) }
+						</Button>
```

Comment:
Disable button when only one checkmark is checked or the selected rows are already being compared?

Reply:
Just looking at the designs and I see it might be something to leave for later

Reply:
Yeah, this still needs to be done on the Compare filter itself too #364 - I figured that could be done with a future PR